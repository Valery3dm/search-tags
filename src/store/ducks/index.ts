import { combineReducers } from 'redux';
import { itemReducer } from './itemReducer';

export const rootReducer = combineReducers({
    item: itemReducer
})

export type RootState = ReturnType<typeof rootReducer>
