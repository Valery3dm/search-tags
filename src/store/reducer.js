const initialState = {
  itemsList: [],
  listOfThreeLastItems: [],
  inputItem: ''
};

const FETCH_ITEMS = "FETCH_ITEMS";
const SET_THREE_LAST_ITEMS = "SET_THREE_LAST_ITEMS";
const SET_INPUT_ITEM = "SET_INPUT_ITEM";

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_ITEMS:
      return { ...state, itemsList: action.payload };
    case SET_THREE_LAST_ITEMS:
      return { ...state,
         listOfThreeLastItems: [ ...state.listOfThreeLastItems, action.payload]
        };
    case SET_INPUT_ITEM:
      return { ...state, inputItem: action.payload };
    default:
      return state;
  }
};

export default reducer;