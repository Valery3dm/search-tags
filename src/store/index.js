import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import createSagaMiddleware from "redux-saga";

import { duck } from "./ducks/widgets";


const sagaMiddleware = createSagaMiddleware();

const store = createStore(duck.reducers.reducer, composeWithDevTools(applyMiddleware(sagaMiddleware)));

export default store;

sagaMiddleware.run(duck.saga);
