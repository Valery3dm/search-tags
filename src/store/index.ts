import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import { sagaFetchItemsWatcher } from './saga/sagaFetchItems';
import { itemReducer } from './ducks/itemReducer';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  itemReducer,
  composeWithDevTools(applyMiddleware(sagaMiddleware))
);

export default store;

sagaMiddleware.run(sagaFetchItemsWatcher);