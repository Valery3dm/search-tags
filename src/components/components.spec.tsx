import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';

import { SearchPanel, PreLoader, LastSearch, Cards } from './index';
import store from '../store';


Enzyme.configure({ adapter: new Adapter() });


describe('components component', () => {
    const listOfThreeLastItems = ['item1','item2','item3'];
    const itemList = [
        {
            id: 1,
            largeImageURL: 'qweasdzxc123',
            tags: 'item1'
        },
        {
            id: 2,
            largeImageURL: 'asdzxc123',
            tags: 'item2'
        },
        {
            id: 3,
            largeImageURL: 'zxc123',
            tags: 'item3'
        }
    ];

    it('SearchPanel Snapshot component', () => {
        const component = shallow(<Provider store={store}><SearchPanel /></Provider>);
        expect(component).toMatchSnapshot();
    });
    it('PreLoader Snapshot component', () => {
        const component = shallow(<PreLoader />);
        expect(component).toMatchSnapshot();
    });
    it('LastSearch Snapshot component', () => {
        const component = shallow(<Provider store={store}><LastSearch listOfThreeLastItems={listOfThreeLastItems}/></Provider>);
        expect(component).toMatchSnapshot();
    });
    it('Cards Snapshot component', () => {
        const component = shallow(<Provider store={store}><Cards itemList={itemList}/></Provider>);
        expect(component).toMatchSnapshot();
    });
});

