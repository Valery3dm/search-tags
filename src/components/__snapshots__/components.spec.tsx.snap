// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components component Cards Snapshot component 1`] = `ShallowWrapper {}`;

exports[`components component LastSearch Snapshot component 1`] = `ShallowWrapper {}`;

exports[`components component PreLoader Snapshot component 1`] = `ShallowWrapper {}`;

exports[`components component SearchPanel Snapshot component 1`] = `ShallowWrapper {}`;
